pyro/optim/optim.py

```diff
@@ -52,8 +66,12 @@ def __call__(self, params,  *args, **kwargs):
             if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler) or \
                     isinstance(self.optim_objs[p], torch.optim.lr_scheduler.ReduceLROnPlateau):
                 # if optim object was a scheduler, perform an optimizer step
+                if self.grad_clip[p] is not None:
```

Comment:
nit: Can you move this outside of the `if isinstance(...optim...)` statement, since the two branches are identical?

Reply:
Oh duh good point! My original version needed to be inside the `if` statement but then later rewrite made it unnecessary